code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -34,124 +31,257 @@
+		return
 
-	playsound(owner, 'sound/voice/predalien_growl.ogg', 75, 0, status = 0)
+	if(predalien_smash.can_not_harm(affected_atom) || !ismob(affected_atom))
+		apply_cooldown_override(click_miss_cooldown)
+		return
 
-	REMOVE_TRAIT(xeno, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Devastate"))
-	xeno.anchored = FALSE
-	unroot_human(carbon, TRAIT_SOURCE_ABILITY("Devastate"))
+	var/mob/living/carbon/carbon = affected_atom
+	if(carbon.stat == DEAD)
+		return
+
+	if (!check_and_use_plasma_owner())
+		return
 
-	xeno.visible_message(SPAN_XENODANGER("[xeno] rapidly slices into [carbon]!"))
+	apply_cooldown()
 
+	predalien_smash.throw_atom(get_step_towards(affected_atom, predalien_smash), grab_range, SPEED_FAST, predalien_smash)
+
+	if (predalien_smash.Adjacent(carbon))
+		predalien_smash.start_pulling(carbon,1)
+		if(isliving(carbon))
+			playsound(carbon.pulledby, 'sound/voice/predalien_growl.ogg', 75, 0, status = 0) // bang and roar for dramatic effect
+			playsound((carbon), 'sound/effects/bang.ogg', 25, 0)
+			animate(carbon, pixel_y = carbon.pixel_y + 32, time = 4, easing = SINE_EASING)
+			sleep(4)
+			playsound((carbon), 'sound/effects/bang.ogg', 25, 0) // bang for dramatic damage effect
+			playsound((carbon),"slam", 50, 1)
+			animate(carbon, pixel_y = 0, time = 4, easing = BOUNCE_EASING)
+			var/turf/back_to_middle = get_step(carbon, SOUTH) // move them back one tile south
+			carbon.forceMove(back_to_middle)
+			carbon.apply_armoured_damage(get_xeno_damage_slash(carbon, smash_damage + smash_scale * predalienbehavior.kills), ARMOR_MELEE, BRUTE, "chest", 20)
+	else
+		predalien_smash.visible_message(SPAN_XENOWARNING("[predalien_smash]'s claws twitch."), SPAN_XENOWARNING("We couldn't grab our target. Wait a moment to try again."))
+
+	return TRUE
+
+/mob/living/carbon/xenomorph/predalien/stop_pulling()
+	if(isliving(pulling) && smashing)
+		smashing = FALSE // To avoid extreme cases of stopping a lunge then quickly pulling and stopping to pull someone else
+		var/mob/living/smashed = pulling
+		smashed.set_effect(0, STUN)
+		smashed.set_effect(0, WEAKEN)
 	return ..()
+
+/mob/living/carbon/xenomorph/predalien/start_pulling(atom/movable/movable_atom, feral_smash)
+	if(!check_state())
+		return FALSE
+
+	if(!isliving(movable_atom))
+		return FALSE
+	var/mob/living/living_mob = movable_atom
+	var/should_neckgrab = !(src.can_not_harm(living_mob)) && feral_smash
+
+
+	. = ..(living_mob, feral_smash, should_neckgrab)
+
+	if(.) //successful pull
+		if(isxeno(living_mob))
+			var/mob/living/carbon/xenomorph/xeno = living_mob
+			if(xeno.tier >= 2) // Tier 2 castes or higher immune to warrior grab stuns
+				return
+
+		if(should_neckgrab && living_mob.mob_size < MOB_SIZE_BIG)
+			visible_message(SPAN_XENOWARNING("[src] grabs [living_mob] by the back of their leg! and repeatedly slams them onto the ground!"), \
+			SPAN_XENOWARNING("We grab [living_mob] by the back of their leg! and repeatedly slam them onto the ground!")) // more flair
```

Comment:
This message should be with the slam code /datum/action/xeno_action/activable/feral_smash/use_ability

Also its wording is weird. The ! is placed in the middle of the sentence and we don't repeatedly slam them:

```suggestion
			visible_message(SPAN_XENOWARNING("[src] grabs [living_mob] by the back of their leg and slams them onto the ground!"), \
			SPAN_XENOWARNING("We grab [living_mob] by the back of their leg and slam them onto the ground!")) // more flair
```